final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,143 @@
+import feedparser
+import sys
+import datetime
+import argparse
+import json
+from bs4 import BeautifulSoup
+
+DEFAULT_LIMIT_COUNT = 3
+VERSION = '1.0'
+
+
+class RSSReader:
+    is_verbose = False  # logging is disabled by default
+    is_json = False
+    limit = DEFAULT_LIMIT_COUNT
+
+    def __init__(self, url):
+        self.url = url
+
+    def url_parsing(self):
+        parsed_url = feedparser.parse(self.url)
+        if parsed_url.status != 200:  # URL-access check
+            print("Can't load RSS feed.")
+            raise Exception('RSS-reader failed. Status: {}.'.format(parsed_url.status))
+        else:
+            self.log('RSS parsed successfully')
+            return parsed_url
+
+    def information_about_site(self, parsed_url):
+        """Function which make dictionary with data of website"""
+        return {
+            'Feed': parsed_url['feed']['title'],
+            'Updated': parsed_url['updated'],
+            'Version': parsed_url['version'],
+        }
+
+    def make_news_data(self, news):
+        """
+        Function which parsed HTML summary of news and make dictionary with all the necessary news data.
+        :return dictionary
+        """
+        bs = BeautifulSoup(news['summary'], 'html.parser')
+        img = bs.find_all('img')
+        image_data = 'No image'
+        if len(img) > 0:
+            image_data = '{}\nSource of image: {}'.format(img[0]['alt'], img[0]['src'])
+        news_summary = {
+            'Title': news['title'],
+            'Date': news['published'],
+            'Summary': bs.get_text(),
+            'Link': news['link'],
+            'Image': image_data,
+        }
+        return news_summary
+
+    def news_data_collection(self, parsed_url):
+        """Function that collects data from all news by calling make_news_data.
+        :return string of dictionaries
+        """
+        all_information_about_news = parsed_url['entries']
+        # amount_of_news = len(parsed_url.entries)
+        all_news = []
+        self.log('News gathering')
+        for news in all_information_about_news[:self.limit]:
+            try:  # Exception when news is failed
+                dictionary_of_news_data = self.make_news_data(news)
+            except Exception as ex:
+                self.log("Error processing news: {} {}".format(type(ex), ex))
+                continue  # continues to process the following news
+            else:
+                all_news.append(dictionary_of_news_data)  # make list of dictionaries of news data for optional output
+        return all_news
+
+    def log(self, message):
+        """Print time and log message if verbose is active."""
+        if self.is_verbose:
+            print(datetime.datetime.now(), message)
```

Comment:
A creative solution, but nevertheless you should better check `logging` standart library. Your current method is hardly scalable and will cause problems for you in future iterations